### ISSUE TITLE
Consider adding additional configuration utilising nixpkgs security/misc.nix module

### ISSUE THREAD

**Issue by wyyllou** [cynicsketch/nix-mineral#17]
https://github.com/NixOS/nixpkgs/blob/nixos-24.05/nixos/modules/security/misc.nix

This would be things like 

```nix
security = {
	virtualisation.flushL1DataCache = true;
	forcePageTableIsolation  = true;
	allowSimultaneousMultithreading = true;
	protectKernelImage = true;
	unprivilegedUsernsClone = true;
	allowUserNamespaces = true;
};
```

allowSimultaneousMultithreading is similar to the implementation at https://github.com/cynicsketch/nix-mineral/blob/main/nix-mineral.nix#L319

**Comment by wyyllou** [comment]
I believe you also in part implemented something of relevance to the allowUserNamespaces option(?)

**Comment by cynicsketch** [comment]
No significant benefit to be seen here, but if you can make it happen then we might as well.

**Comment by wyyllou** [comment]
security.protectKernelImage would provide the most benefit here since i don't believe you have implemented this

**Comment by cynicsketch** [comment]
> `security.protectKernelImage` would provide the most benefit here since i don't believe you have implemented this (or even can easily)

I don't really like this, for a couple reasons:

1. NixOS [doesn't use kernel live patching](https://github.com/NixOS/nixpkgs/issues/10726) so nobody should be caring (as of present) that kexec is disabled by default when there's no way could possibly use it even if it was enabled.
2. `security.protectKernelImage` implicitly disables hibernation with the boot parameter `nohibernate`. This stems from the logic seen [here](https://unix.stackexchange.com/questions/591488/why-does-the-kernel-lockdown-prevent-hibernation).
 
In that case, it makes more sense to just use lockdown or, in the rare case that someone [doesn't want enforced module signing](https://patchwork.kernel.org/project/linux-security-module/patch/20190820001805.241928-5-matthewgarrett@google.com/) but is also okay with disabling hibernation, separating the option makes more sense.

Basically, it's a granularity issue.

**Comment by wyyllou** [comment]
ok, i think it would make sense to close this issue then, and the user can choose since they are built in nixpkgs options.
